lib/src/main/kotlin/com/swmansion/starknet/provider/gateway/GatewayProvider.kt

```diff
@@ -106,6 +103,18 @@ class GatewayProvider(
             json.decodeFromString(deserializationStrategy, body)
         }
 
+    private fun transformResponseForTransactionAndSerialize(response: HttpResponse): Transaction {
+        val body = handleResponseError(response)
+
+        try {
+            val transformedResponse = json.parseToJsonElement(body).jsonObject["transaction"]
+            val encodedResponse = json.encodeToString(transformedResponse)
+            return json.decodeFromString(TransactionPolymorphicSerializer, encodedResponse)
```

Comment:
There is most probably, I experimented a lot with combining multiple serializer in order to avoid this, but instead of answers and solutions I got a headache. At the end decided I burned too much time on this, and went with this instead. If you don't like it, I can maybe create an issue to deal with it later?

Reply:
Sure, let's make an issue for this, it's not very critical.

Reply:
https://github.com/software-mansion/starknet-jvm/issues/220